### ISSUE TITLE
[question] Project manifest doesn't match installed one

### ISSUE THREAD

**Issue by mnesarco** [conan-io/conan#12104]
Hello Friends,
I am trying to build opencascade locally from the recipe. It looks like everything works, but at the end I get this error:

```
ERROR: New manifest 'brotli/1.0.9' detected.
Remote: conancenter:https://center.conan.io
Project manifest doesn't match installed one
```

I tried to find information about this error but I have found nothing.

This is my command:
```
conan create . opencascade-marz/7.7.0@mnesarco/dev --profile default -v --update --build=missing
```

The recipe was downloaded from conan index, I added the sources url, sha256 for version 7.7.0. and commented out a line in conanfile.py just ignoring a replacement in file that does not apply to master.

https://github.com/conan-io/conan-center-index/tree/master/recipes/opencascade

my conan version is:
```
Conan version 1.52.0
```

my plaform:
```
Linux Mint 21
5.15.0-47-generic #51-Ubuntu SMP x86_64 GNU/Linux
```

my profile:
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=clang
compiler.version=14
compiler.libcxx=libc++
build_type=Release

[options]

[build_requires]

[env]
CC=/usr/bin/clang
CXX=/usr/bin/clang++

```

All dependencies are downloaded and built. Opencascade is built apparently because I can see the message:

```
...
opencascade-marz/7.7.0@mnesarco/dev: Package 'fe8e9f78b7cfc07ca05fddec15478bf3a4cc8c9b' created
opencascade-marz/7.7.0@mnesarco/dev: Created package revision 53da2e431f4356d575c9b05eee63fa4b
opencascade-marz/7.7.0@mnesarco/dev: Appending PATH environment variable: /xxxxx/.conan/data/opencascade-marz/7.7.0/mnesarco/dev/package/fe8e9f78b7cfc07ca05fddec15478bf3a4cc8c9b/bin
Downloading conan_sources.tgz completed [2.89k]                                          
Decompressing conan_sources.tgz completed [0.00k]                                        
Downloading conan_sources.tgz completed [1.83k]                                          
Decompressing conan_sources.tgz completed [0.00k]                                        
Downloading conan_sources.tgz completed [10.58k]                                         
Decompressing conan_sources.tgz completed [0.00k]                                        
Downloading conan_sources.tgz completed [0.40k]                                          
Decompressing conan_sources.tgz completed [0.00k]                                        
Downloading conan_sources.tgz completed [1.17k]                                          
Decompressing conan_sources.tgz completed [0.00k]                                        
ERROR: New manifest 'brotli/1.0.9' detected.
Remote: conancenter:https://center.conan.io
Project manifest doesn't match installed one

```

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by mnesarco** [comment]
This is my first usage of conan. It is hard. I can build opencascade from sources directly without problem but I expect that conan helps me doing this much more easy.

what I have learned:
- Conan is hard.
- The problem **"Project manifest doesn't match installed one"** is apparently related to `test_package` because if I set` --test-folder=None` there are no errors.

Now the package is generated and deployed to the local cache:
```
Existing package recipes:

autoconf/2.71
automake/1.16.5
brotli/1.0.9
bzip2/1.0.8
expat/2.4.8
fontconfig/2.13.93
freetype/2.11.1
freetype/2.12.1
gnu-config/cci.20210814
gperf/3.1
libpng/1.6.37
libtool/2.4.7
libuuid/1.0.3
m4/1.4.19
opencascade-marz/7.7.0
opencascade-marz/7.7.0@mnesarco/dev
opengl/system
pkgconf/1.7.4
tcl/8.6.11
tk/8.6.10
xorg/system
zlib/1.2.12
```

but the **test_package** does not build:

```
mkdir build
cd build
conan install ..

... lots of files generated, no errors
```

then cmake fails:

```
[main] Building folder: test_package 
[build] Starting build
[proc] Executing command: /xxx/devel-tools/cmake/current/bin/cmake --build /xxx/devel/conan/opencascade/all/test_package/build --config Debug --target all --
[build] [1/2  50% :: 0.234] Building CXX object CMakeFiles/test_package.dir/test_package.cpp.o
[build] FAILED: CMakeFiles/test_package.dir/test_package.cpp.o 
[build] /usr/bin/clang++   -g -std=gnu++20 -MD -MT CMakeFiles/test_package.dir/test_package.cpp.o -MF CMakeFiles/test_package.dir/test_package.cpp.o.d -o CMakeFiles/test_package.dir/test_package.cpp.o -c /xxx/devel/conan/opencascade/all/test_package/test_package.cpp
[build] /xxx/devel/conan/opencascade/all/test_package/test_package.cpp:1:10: fatal error: 'BRepBuilderAPI_MakeEdge.hxx' file not found
[build] #include <BRepBuilderAPI_MakeEdge.hxx>
[build]          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[build] 1 error generated.
[build] ninja: build stopped: subcommand failed.
[proc] The command: /xxx/devel-tools/cmake/current/bin/cmake --build /xxx/devel/conan/opencascade/all/test_package/build --config Debug --target all -- exited with code: 1 and signal: null
[build] Build finished with exit code 1

```
This is the conanfile.py of test_package (copied from the original recipe with added requirements):

```python
from conans import ConanFile, CMake, tools
import os


class TestPackageConan(ConanFile):
    settings = "os", "arch", "compiler", "build_type"
    generators = "cmake", "cmake_find_package_multi"

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

    def test(self):
        if not tools.cross_building(self):
            bin_path = os.path.join("bin", "test_package")
            self.run(bin_path, run_environment=True)
            bin_path_global = os.path.join("bin", "test_package_global")
            self.run(bin_path_global, run_environment=True)

    def requirements(self):
        self.requires("opencascade-marz/7.7.0@mnesarco/dev")

```

**Comment by mnesarco** [comment]
I managed to make it work changing the CMakeLists.txt of the test_package:
```

cmake_minimum_required(VERSION 3.1)
project(test_package CXX)

set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH};${CMAKE_BINARY_DIR})

include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
conan_basic_setup(TARGETS)

# Test components
find_package(OpenCASCADE COMPONENTS FoundationClasses ModelingData ModelingAlgorithms REQUIRED)

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} TKMath TKGeomBase TKTopAlgo TKBRep)
set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 20)
 
# Test global conan target
# add_executable(${PROJECT_NAME}_global test_package.cpp)
# target_link_libraries(${PROJECT_NAME}_global CONAN_PKG::opencascade)
# set_property(TARGET ${PROJECT_NAME}_global PROPERTY CXX_STANDARD 11)
```

The change required is a bit weird because I expect that `conanbuildinfo.cmake` take care of everything. But no.

Required change:
`set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH};${CMAKE_BINARY_DIR})`

**Comment by mnesarco** [comment]
Hello Friends,
As a new comer the experience was horrible at the beginning. But now that I  have managed to setup `conan`+`cmake`+`ninja`+`clang`+`vscode` I am converted.

I removed conanfile.py and I am using only conanfile.txt now:

```
[requires]
opencascade-dev/7.7.0@mnesarco/dev
fmt/9.1.0
glfw/3.3.8

[generators]
CMakeDeps
CMakeToolchain
```

And my default profile:

```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=clang
compiler.version=14
compiler.libcxx=libc++
build_type=Release

[options]

[build_requires]

[env]
CC=/usr/bin/clang
CXX=/usr/bin/clang++
```

The most helpful resource was this video: https://youtu.be/X4bPsRAXHRs

Many thanks for all the developers and maintainers behind conan.

**Comment by memsharded** [comment]
Hi @mnesarco 

Thanks for your feedback and following up. Happy to hear that you are converted now.
I am sorry that your onboarding experience was not as smooth as we intended, but let me clarify a few factors that could help explaining a bit:

- We are in the middle of the transition of 1.X->2.0, and this is a huge release, with many changes. The docs are a bit messy right now, we are re-writing them from scratch for 2.0
- Recipes in ConanCenter are built and tested for mainstream compilers, sometimes can be a bit rough for other compilers. Clang in Linux should be relatively ok, but to be taken into account for others.
- Recipes in ConanCenter are also moving forward to be ready for 2.0, quickly, and they can be also a bit bumpy
- New build system integrations for 2.0, backported to 1.X like CMakeToolchain, CMakeDeps also change and improve release to release, recently adding CMakePresets for example, but integrations with IDEs are still a bit rough, and very importantly, not enough documented.
- Also we have been a quite busy (conference season), we usually are able to respond to questions within a couple of days, and try to support users better.

We are really confident that Conan 2.0 will be much, much better than 1.X, both in the tool itself, but also in the docs, which is something core to a good onboarding experience. Thanks for all the feedback!

**Comment by mnesarco** [comment]
Thank you @memsharded ,
Yes I think the main problem was confusing/missing documentation.

My use case was:
1. Build a package locally using a custom recipe (copy of existing one with minor updates)
2. Use the local package.

There was obscure errors, hard to understand and hard to find info about them. I saw the light at the end of the tunnel by chance in a Youtube video.

In the end, everything worked with a simple text file, so the tool is powerful. But requires better docs.

In retrospective, I think in my case the problem was something wrong in the test_package of the original recipe. I have no idea what is wrong there and I don't care. `"Project manifest doesn't match installed one"` is an undecipherable error message for me at this point, so I built it with `--test-folder=None` and moved on.

All these things may be obvious for an experienced conan user for sure. on-boarding on anything related to C++ is hard.

Thank you for all the hard work, I know you are trying to make things easier for the rest of us.
